### PR TITLE
480/Add intent extras for custom tabs

### PR DESCRIPTION
**Pull request by friedger**
When opening an external link we add now custom tabs information for color and title.

Before | After
---- | -----
 |

### CODE REVIEW

**Comment by tasomaniac** [comment]
Does this still ask the user to open the link in native Google+ app? I think opening these in Google+ app has much better experience.

File: app/src/main/java/org/gdg/frisbee/android/utils/Utils.java [hunk 1]
```diff
@@ -218,7 +219,12 @@ public static boolean canLaunch(Context context, final Intent viewUrlIntent) {
 
     @SuppressWarnings("deprecation")
     public static Intent createExternalIntent(Context context, Uri uri) {
-        Intent intent = new Intent(Intent.ACTION_VIEW, uri);
+        CustomTabsIntent customTabsIntent = new CustomTabsIntent.Builder()
+                .setToolbarColor(context.getResources().getColor(R.color.theme_primary))
```
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/utils/Utils.java]: You may also use this to access real primaryColor of the theme. I am not really sure which is the best. We are using different colorPrimary for DevFest, StudyJams etc. 

        ThemeUtils.getThemeAttrColor(context, R.attr.colorPrimary);

File: app/src/main/java/org/gdg/frisbee/android/utils/Utils.java [hunk 2]
```diff
@@ -218,7 +219,12 @@ public static boolean canLaunch(Context context, final Intent viewUrlIntent) {
 
     @SuppressWarnings("deprecation")
     public static Intent createExternalIntent(Context context, Uri uri) {
-        Intent intent = new Intent(Intent.ACTION_VIEW, uri);
+        CustomTabsIntent customTabsIntent = new CustomTabsIntent.Builder()
+                .setToolbarColor(context.getResources().getColor(R.color.theme_primary))
+                .setShowTitle(true)
+                .build();
+        Intent intent = customTabsIntent.intent;
+        intent.setData(uri);
```
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/utils/Utils.java]: Don't we have to call `launchUrl` function of customTabs to make it work?

**Comment by friedger** [comment]
The intent is not changed, only extras added. Same experience as before for all apps besides chrome.

File: app/src/main/java/org/gdg/frisbee/android/utils/Utils.java (hunk 2 above)
Comment (friedger) [inline app/src/main/java/org/gdg/frisbee/android/utils/Utils.java]: No, launchUrl is just a convenience method

**Comment by tasomaniac** [comment]
It is nice that nothing is changing. It can open Google+ like before. I didn't know it is just Intent Extras. Service binding is only for accessing internals and warm up. 

I've added this flag `FLAG_ACTIVITY_CLEAR_WHEN_TASK_RESET` before to display Google+ on its own task. Because of this CustomTabs look like outside of our Activity. Recent screen looks like this:

device-2015-09-05-012721

You see Chrome logo is showing. :/

**Comment by tasomaniac** [comment]
I've removed `FLAG_ACTIVITY_CLEAR_WHEN_TASK_RESET` flag previously added by me. I hope you don't mind. If you can briefly review my commit we can merge it.

**Comment by tasomaniac** [comment]
@friedger What do you think of the change I've committed? If it is ok feel free to merge this, if not feel free to revert it. :smile:

**Pull request merged**